resources/indexer/directory_scanner.py

```diff
@@ -0,0 +1,63 @@
+"""Recursively scan directory paths and print all file paths."""
+
+import argparse
+import logging
+import os
+from concurrent.futures import ProcessPoolExecutor
+from time import sleep
+
+
+def process_dir(path):
+    """Print out file paths and return sub-directories."""
+    try:
+        scan = list(os.scandir(path))
+    except (PermissionError, FileNotFoundError):
+        scan = []
+    dirs = []
+
+    all_file_count = 0
+    for dir_entry in scan:
+        if dir_entry.is_symlink():
```

Comment:
Should also check for other types of files, like pipes or devices. Those should be skipped as well.

Reply:
I'm looking at https://docs.python.org/3/library/pathlib.html#pathlib.Path.is_dir. Do you know if there's a type hierarchy here, ie. `(path.is_file() and path.is_fifo()) is True`?

Reply:
Most of them are probably also files, but otherwise there isn't much of a hierarchy.

Reply:
My read of the documentation is that each of those are orthogonal, with the exception that a symlink pointing to a regular file will be true for both `is_file` and `is_symlink`. Otherwise it should be a choose-one between them.

If this is how it actually works in practice... :man_shrugging: you may need to write some unit tests, depending on how important it is.

Reply:
The key ones are socket and fifo, so just adding them to the if statement should be fine.